### PR TITLE
Dialog Api improvements

### PR DESCRIPTION
**Pull request by sean-brydon**
Cleanup of how we handle dialogs.

Moved to using children approach instead of passing actions via props. 

Follows practises seen in MUI/Chakra

### CODE REVIEW

File: packages/ui/v2/core/Dialog.tsx [hunk 1]
```diff
@@ -175,12 +141,32 @@ export function DialogHeader(props: DialogHeaderProps) {
 export function DialogFooter(props: { children: ReactNode }) {
   return (
     <div>
-      <div className="mt-5 flex justify-end space-x-2 rtl:space-x-reverse">{props.children}</div>
+      <div className="mt-7 flex justify-end space-x-2 rtl:space-x-reverse ">{props.children}</div>
```
Comment (sean-brydon) [inline packages/ui/v2/core/Dialog.tsx:144]: We were often wrapping the child of dialog footer with mt-2 so I just merged it here.

File: packages/ui/v2/core/Dialog.tsx [hunk 2]
```diff
@@ -175,12 +141,32 @@ export function DialogHeader(props: DialogHeaderProps) {
...
+  const { t } = useLocale();
+  return (
+    <DialogPrimitive.Close asChild {...props.dialogCloseProps}>
+      {/* This will require the i18n string passed in */}
+      <Button color={props.color || "minimal"} {...props}>
+        {props.children ? props.children : t("Close")}
+      </Button>
+    </DialogPrimitive.Close>
+  );
+}
```
Comment (sean-brydon) [inline packages/ui/v2/core/Dialog.tsx:172]: Create a wrapper for dialogclose - no point doing 
```
<DialogClose>
	<Button />
</DialogClose> 
```
Every single time we need this

File: packages/ui/v2/modules/event-types/CreateEventType.tsx [hunk 3]
```diff
@@ -315,9 +314,7 @@ export default function CreateEventTypeButton(props: CreateEventTypeBtnProps) {
             <Button type="submit" loading={createMutation.isLoading}>
               {t("continue")}
             </Button>
-            <DialogClose asChild>
-              <Button color="secondary">{t("cancel")}</Button>
-            </DialogClose>
+            <DialogClose />
```
Comment (sean-brydon) [inline packages/ui/v2/modules/event-types/CreateEventType.tsx:317]: Much cleaner ;)

File: packages/ui/TimezoneChangeDialog.tsx [hunk 4]
```diff
@@ -58,20 +59,23 @@ export default function TimezoneChangeDialog() {
 
   if (data?.user.impersonatedByUID) return null;
 
+  const ONE_DAY = 60 * 60 * 24; // 1 day in seconds (60 seconds * 60 minutes * 24 hours)
```
Comment (JeroenReumkens) [inline packages/ui/TimezoneChangeDialog.tsx:62]: 🥇

File: packages/ui/v2/modules/event-types/CreateEventType.tsx (hunk 3 above)
Comment (JeroenReumkens) [reply packages/ui/v2/modules/event-types/CreateEventType.tsx:317]: MUCH!

**Review by JeroenReumkens — APPROVED** [review]
Well done! So much better 👏

**Review by PeerRich — APPROVED** [review]

**Comment by JeroenReumkens** [comment]
Ready for merge. waiting for a merge from main which fixes the E2E tests.

**Comment by PeerRich** [comment]
@JeroenReumkens or @zomars E2E is failing https://github.com/calcom/cal.com/actions/runs/3564670650/jobs/5988914590 can one of you investigate so we can merge this? 

lets also tag a release pretty soon

File: apps/web/pages/settings/my-account/profile.tsx [hunk 5]
```diff
@@ -298,13 +300,7 @@ const ProfileView = () => {
             title={t("delete_account_modal_title")}
             description={t("confirm_delete_account_modal")}
             type="creation"
-            actionText={t("delete_my_account")}
-            actionProps={{
-              // @ts-expect-error data attributes aren't typed
-              "data-testid": "delete-account-confirm",
```
Comment (zomars) [inline apps/web/pages/settings/my-account/profile.tsx:304]: Is this being used in a test?

Comment (sean-brydon) [reply apps/web/pages/settings/my-account/profile.tsx:304]: Yes but it also being passed into the button on line 328 so im not 100% sure what's happening here

**Pull request merged**